### PR TITLE
Log whether PTDS is enabled

### LINKED ISSUE #619: [FEA] log whether libcudf and libcudfjni were built for PTDS
Given queriable flags defined in:

- CUDF JNI: https://github.com/rapidsai/cudf/issues/6112
- CUDF: https://github.com/rapidsai/cudf/issues/6111

Produce a log at startup stating whether the plugin is running with PTDS enabled. Additionally, if there is a mismatch between these two flags (jni/libcudf) we should ERROR at least, and one could argue we show blow up... since we have a PTDS and not-PTDS.

### PR DESCRIPTION
**Pull request by rongou**
Closes #619 

@jlowe @abellina

### CODE REVIEW

**Comment by abellina** [comment]
We do quite a bit of logging here: https://github.com/NVIDIA/spark-rapids/blob/branch-0.3/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala#L198, especially around RMM. I'd suggest adding the line? Maybe in the same line where we state how much we allocated via RMM.

**Comment by tgravescs** [comment]
yeah you don't want to use cuda at this point on the driver as it will require driver to have gpu and driver.  It would be better in the GpuDeviceManager as Alessandro suggests.

**Comment by rongou** [comment]
@abellina @tgravescs done.

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala [hunk 1]
```diff
@@ -21,6 +21,7 @@ import java.util.concurrent.atomic.{AtomicBoolean, AtomicReference}
 
 import scala.collection.JavaConverters._
 
+import ai.rapids.cudf.Cuda
```
Comment (abellina) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala:24]: Extra import, we can remove this one

Comment (rongou) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala]: Done.

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Comment by jlowe** [comment]
CI failed because the new cudf API is not in 0.16.  We won't be able to merge this PR until the plugin moves to cudf 0.17-SNAPSHOT.

**Review by tgravescs — APPROVED** [review]

**Review by abellina — APPROVED** [review]

**Comment by jlowe** [comment]
@rongou now that #1040 has been merged, this PR can be updated to latest on branch-0.3 and that should allow the premerge CI to run.

**Comment by rongou** [comment]
build

**Pull request merged**